javascripts/discourse/lib/docs-sidebar-panel.js

```diff
@@ -0,0 +1,197 @@
+import { cached } from "@glimmer/tracking";
+import { computed } from "@ember/object";
+import { htmlSafe } from "@ember/template";
+import BaseCustomSidebarSection from "discourse/lib/sidebar/base-custom-sidebar-section";
+import BaseCustomSidebarSectionLink from "discourse/lib/sidebar/base-custom-sidebar-section-link";
+import DiscourseURL from "discourse/lib/url";
+import { unicodeSlugify } from "discourse/lib/utilities";
+import { getOwnerWithFallback } from "discourse-common/lib/get-owner";
+import getURL, {
+  getAbsoluteURL,
+  samePrefix,
+} from "discourse-common/lib/get-url";
+import I18n from "discourse-i18n";
+import { SIDEBAR_DOCS_PANEL } from "../services/docs-sidebar";
+
+const sidebarPanelClassBuilder = (BaseCustomSidebarPanel) =>
+  class DocsSidebarPanel extends BaseCustomSidebarPanel {
+    key = SIDEBAR_DOCS_PANEL;
+    hidden = true;
+    displayHeader = true;
+
+    get docsSidebar() {
+      return getOwnerWithFallback(this).lookup("service:docs-sidebar");
+    }
+
+    @cached
+    get sections() {
+      const router = getOwnerWithFallback(this).lookup("service:router");
+
+      return this.docsSidebar.sectionsConfig.map((config) => {
+        return prepareDocsSection({ config, router });
+      });
+    }
+
+    get filterable() {
+      return !this.docsSidebar.loading;
+    }
+
+    filterNoResultsDescription(filter) {
+      const active = this.docsSidebar.activeCategory;
+      let categoryFilter = "";
+
+      if (this.docsSidebar.activeCategory) {
+        categoryFilter =
+          " " +
+          (this.#assembleCategoryFilter("", active, 1) ??
+            `category:${active.id}`);
+      }
+
+      const params = {
+        filter,
```

Comment:
We need to `escapeExpression` this, because it's about to be interpolated into an htmlSafe string.